validation/rules.go

```diff
@@ -0,0 +1,26 @@
+package validation
+
+import "fmt"
+
+const (
+	// UDP protocol
+	UDP = `UDP`
+	// TCP protocol
+	TCP = `TCP`
+)
+
+// IsValidPortNum tests that the argument is a valid, non-zero port number.
+func IsValidPortNum(port int) error {
+	if 1 <= port && port <= 65535 {
+		return nil
+	}
+	return fmt.Errorf(`must be between %d and %d, inclusive`, 1, 65535)
+}
+
+// IsValidProtocol tests that the argument is TCP or UDP.
+func IsValidProtocol(protocol string) error {
+	if protocol != TCP && protocol != UDP {
+		return fmt.Errorf(`must be TCP or UDP`)
```

Comment:
It would be nice to be able to fix up the case for the user, but I guess we don't have a good place to do the normalization right now. There's some in [`LoadRoleManifest`](https://github.com/hpcloud/fissile/blob/8fa6780b7c966e26f4bcbb2c76f0dcf34df6ce28/model/roles.go#L196) but that would be cleaner if we moved that around.